### ISSUE TITLE
Example 14 Knot supervised train loss NAN

### ISSUE THREAD

**Issue by gllzjlab** [KindXiaoming/pykan#563]
When I execute the code "Example 14 Knot supervised.py", regardless of how I modify the learning rate or optimizer, the train_loss always remains NaN. I'm using PyTorch version 2.2.2 with Python 3.9.21. Could you please help me resolve this issue? Thank you!
